bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/service/SmartThingsApiService.java

```diff
@@ -0,0 +1,981 @@
+            }
+            return "";
+        }
+
+        public String getInputSource() {
+            if (getCapabilityAttribute("mediaInputSource", "inputSource")
+                    || getCapabilityAttribute("samsungvd.mediaInputSource", "inputSource")) {
+                return Optional.ofNullable(deviceEvent).map(a -> a.getValue()).orElse("");
+            }
+            return "";
+        }
+
+        public String[] getInputSourceList() {
+            if (getCapabilityAttribute("mediaInputSource", "supportedInputSources")) {
+                return deviceEvent.getValuesAsList().toArray(String[]::new);
+            }
+            return new String[0];
+        }
+
+        public List<?> getInputSourceMapList() {
+            if (getCapabilityAttribute("samsungvd.mediaInputSource", "supportedInputSourcesMap")) {
+                return deviceEvent.getValuesAsList();
+            }
+            return List.of();
+        }
+
+        public int getInputSourceId() {
+            return this.tvInfo.map(t -> IntStream.range(0, t.getSources().length)
+                    .filter(i -> t.getSources()[i].equals(getInputSource())).findFirst().orElse(-1)).orElse(-1);
+        }
+
+        public Number getTvChannel() {
+            if (getCapabilityAttribute("tvChannel", "tvChannel")) {
+                return Optional.ofNullable(deviceEvent).map(a -> a.getValue()).filter(i -> !i.isBlank())
+                        .map(j -> parseTVChannel(j)).orElse(-1f);
+            }
+            return -1;
+        }
+
+        public String getTvChannelName() {
+            if (getCapabilityAttribute("tvChannel", "tvChannelName")) {
+                return Optional.ofNullable(deviceEvent).map(a -> a.getValue()).orElse("");
+            }
+            return "";
+        }
+    }
+
+    public Number parseTVChannel(String channel) {
+        try {
+            return Optional.ofNullable(channel)
+                    .map(a -> a.replaceAll("\\D+", ".").replaceFirst("^\\D*((\\d+\\.\\d+)|(\\d+)).*", "$1"))
+                    .map(Float::parseFloat).orElse(-1f);
+        } catch (NumberFormatException ignore) {
+        }
+        return -1;
+    }
+
+    public void updateTV() {
+        if (!tvInfo.isPresent()) {
+            fetchdata();
+            tvInfo.ifPresent(t -> {
+                updateState(CHANNEL_NAME, t.getTvChannelName());
+                updateState(CHANNEL, t.getTvChannel());
+                updateState(SOURCE_NAME, t.getInputSource());
+                updateState(SOURCE_ID, t.getInputSourceId());
+            });
+        }
+    }
+
+    /**
+     * Smartthings API HTTP interface
+     * Currently rate limited to 350 requests/minute
+     *
+     * @param method the method "GET" or "POST"
+     *
+     * @param URI uri
+     *
+     * @param content to POST (or null)
+     *
+     * @return String response
+     */
+    public Optional<String> sendUrl(String method, URI uri, @Nullable InputStream content) throws IOException {
```

Comment:
Better to use an enum in the method signature. Don;t forget to add the import:
```import org.eclipse.jetty.http.HttpMethod;```

```suggestion
    public Optional<String> sendUrl(HttpMethod method, URI uri, @Nullable InputStream content) throws IOException {
```

Reply:
Done